core/chains/solana/soltxm/txm.go

```diff
@@ -0,0 +1,111 @@
+package soltxm
+
+import (
+	"context"
+	"fmt"
+
+	solanaGo "github.com/gagliardetto/solana-go"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana"
+	solanaClient "github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+var (
+	_ services.ServiceCtx = (*Txm)(nil)
+	_ solana.TxManager    = (*Txm)(nil)
+)
+
+// Txm manages transactions for the solana blockchain.
+// simple implementation with no persistently stored txs
+type Txm struct {
+	starter    utils.StartStopOnce
+	lggr       logger.Logger
+	tc         func() (solanaClient.ReaderWriter, error)
+	queue      chan *solanaGo.Transaction
+	stop, done chan struct{}
+	cfg        config.Config
+}
+
+// NewTxm creates a txm. Uses simulation so should only be used to send txes to trusted contracts i.e. OCR.
+func NewTxm(tc func() (solanaClient.ReaderWriter, error), cfg config.Config, lggr logger.Logger) *Txm {
+	lggr = lggr.Named("Txm")
+	return &Txm{
+		starter: utils.StartStopOnce{},
+		tc:      tc,
+		lggr:    lggr,
+		queue:   make(chan *solanaGo.Transaction, 1000), // queue can support 1000 pending txs
+		stop:    make(chan struct{}),
+		done:    make(chan struct{}),
+		cfg:     cfg,
+	}
+}
+
+// Start subscribes to queuing channel and processes them.
+func (txm *Txm) Start(context.Context) error {
+	return txm.starter.StartOnce("solanatxm", func() error {
+		go txm.run()
+		return nil
+	})
+}
+
+func (txm *Txm) run() {
+	defer close(txm.done)
+	ctx, cancel := utils.ContextFromChan(txm.stop)
+	defer cancel()
+	for {
+		select {
+		case tx := <-txm.queue:
+			// TODO: this section could be better optimized for sending TXs quickly
+			// fetch client
+			client, err := txm.tc()
+			if err != nil {
+				txm.lggr.Errorw("failed to get client", "err", err)
+			}
+			// process tx
+			sig, err := client.SendTx(ctx, tx)
+			if err != nil {
+				txm.lggr.Errorw("failed to send transaction", "err", err)
+				continue
+			}
+			txm.lggr.Debugw("successfully sent transaction", "signature", sig)
+		case <-txm.stop:
+			return
+		}
+	}
+}
+
+// TODO: transaction confirmation
+// use ConfirmPollPeriod() in config
+
+// Enqueue enqueue a msg destined for the solana chain.
+func (txm *Txm) Enqueue(accountID string, msg *solanaGo.Transaction) error {
+	select {
+	case txm.queue <- msg:
+	default:
+		txm.lggr.Errorw("failed to enqeue tx", "queueLength", len(txm.queue), "tx", msg)
```

Comment:
How likely and how significant is this failure? Maybe worth `critical` level? On the other hand, if a nodeop can't do anything about it themselves (e.g. network down), then error may be fine.

Reply:
I'd agree with bumping it to `critical`, it should never happen because the tx manager should just keep pulling new txs to send. the only situation i can think that it could happen:
* slow RPC responses (could indicate network load or slow local node, or even network down)